pkg/cmd/server/etcd/etcd.go

```diff
@@ -135,7 +135,7 @@ func TestNewEtcdClient(etcdClient newetcdclient.Client) error {
 // NeverRetryOnFailure is a retry function for the etcdClient.  If there's only one machine, master election doesn't make much sense,
 // so we don't bother to retry, we simply dump the failure and return the error directly.
 func NeverRetryOnFailure(cluster *etcdclient.Cluster, numReqs int, lastResp http.Response, err error) error {
-	if len(cluster.Machines) > 1 {
+	if len(cluster.Machines) > 1 || numReqs < 2 {
```

Comment:
> A single retry on every api call dramatically reducing the impact of this
> flake in test integration tells me that our assumption that leader is
> stable is potentially flawed, since a subsequent request within the WAL
> sync window is also going to get timed out.

Now it wouldn't.  Look at the timeout times.  As I recall its 5 seconds, but previously most failures happen before 10 seconds.  That means that blocking on the sync just meant that we barely finished the retry in time.

Why not gather the actual data from the previous slow location and see if your failures are still coming from there?